codecov_auth/authentication/repo_auth.py

```diff
@@ -398,3 +403,56 @@ def authenticate(
             RepositoryAsUser(repository),
             TokenlessAuth(repository),
         )
+
+
+class UploadTokenRequiredGetFromBodySerializer(serializers.Serializer):
+    slug = serializers.CharField(required=True)
+    service = serializers.CharField(required=False)  # git_service from TA
+    git_service = serializers.CharField(required=False)  # git_service from BA
+
+
+class UploadTokenRequiredGetFromBodyAuthenticationCheck(
+    UploadTokenRequiredAuthenticationCheck
+):
+    """
+    Get Repository and Owner from request body instead of path,
+    then use the same authenticate() as parent class.
+    """
+
+    def _get_git(self, validated_data):
+        """
+        BA sends this in as git_service, TA sends this in as service.
+        Use this function so this Check class can be used by both views.
+        """
```

Comment:
Could probably be `return validated_data.get("git_service") or validated_data.get("service")`